scripts/release.sh

```diff
@@ -0,0 +1,131 @@
+      This new version branch will be a copy of the "current" branch.
+      Default: <undefined>
+  -rn|--remote-name
+      The name of the git remote where the branches are located. This is typically "origin".
+      Default: <undefined>
+  -sb|--staging-branch
+      The name of the branch containing the content that will become the current version of the documentation.
+      Default: "${DEFAULT_STAGING_BRANCH}"
+HELPMSG
+      exit 1
+      ;;
+    *)
+      echo "Unknown argument [$key]. Aborting."
+      exit 1
+      ;;
+  esac
+done
+
+: ${CURRENT_BRANCH:=${DEFAULT_CURRENT_BRANCH}}
+: ${STAGING_BRANCH:=${DEFAULT_STAGING_BRANCH}}
+
+if [ -z "${CURRENT_VERSION:-}" ]; then
+  echo "ERROR! You must specify a --current-version."
+  exit 1
+fi
+
+if [ -z "${REMOTE_NAME:-}" ]; then
+  echo "ERROR! You must specify a --remote-name."
+  exit 1
+fi
+
+# Require vX.Y.Z with Z == 0.
+if ! [[ ${CURRENT_VERSION} =~ ^v[0-9]+\.[0-9]+\.0$ ]]; then
+  echo "ERROR! You must specify --current-version in the form of vX.Y.Z where X and Y are numbers and Z is 0: ${CURRENT_VERSION}"
+  exit 1
+fi
+
+# Only use the vX.Y portion of the version string (strip the patch number).
+# Because the version is used as part of the base image, the version for the base image URL should have dots converted to dashes
+CURRENT_VERSION=$(echo ${CURRENT_VERSION} | sed -E 's/(v[0-9]+\.[0-9]+)\.[0-9]+/\1/')
+CURRENT_VERSION_WITH_DASHES="${CURRENT_VERSION//./-}"
+
+# Prepare the new params.versions to be added to the config.toml within the staging branch
+NEXT_PARAMS_VERSIONS_PLACEHOLDER="## NEXT PARAMS.VERSIONS PLACEHOLDER"
+NEW_PARAMS_VERSIONS=$(cat <<EOM
+${NEXT_PARAMS_VERSIONS_PLACEHOLDER}
+
+[[ params.versions ]]
+  version = "${CURRENT_VERSION}"
+  url = "https://${CURRENT_VERSION_WITH_DASHES}.kiali.io"
+EOM
+)
+
+echo "===== SETTINGS"
+echo CURRENT_BRANCH=$CURRENT_BRANCH
+echo CURRENT_VERSION=$CURRENT_VERSION
+echo CURRENT_VERSION_WITH_DASHES=$CURRENT_VERSION_WITH_DASHES
+echo REMOTE_NAME=$REMOTE_NAME
+echo STAGING_BRANCH=$STAGING_BRANCH
+printf "NEW_PARAMS_VERSIONS:\n${NEW_PARAMS_VERSIONS}\n"
+echo "===== SETTINGS"
+
+echo "===== Fetch the remote content"
+git fetch ${REMOTE_NAME}
+
+echo "===== Create a new version branch named [${CURRENT_VERSION}] based on branch [${CURRENT_BRANCH}]"
+git checkout -b ${CURRENT_VERSION} ${REMOTE_NAME}/${CURRENT_BRANCH}
+
+echo "===== Set baseURL in config.toml for the branch [${CURRENT_VERSION}]"
+sed -i "s/baseURL = .*/baseURL = \"https:\/\/${CURRENT_VERSION_WITH_DASHES}.kiali.io\"/" config.toml
+git commit -am "Set base URL for branch: ${CURRENT_VERSION}"
+
+echo "===== Push the new version branch [${CURRENT_VERSION}] to remote [${REMOTE_NAME}]"
+git push ${REMOTE_NAME} ${CURRENT_VERSION}
+
+echo "===== Create a new branch named [${STAGING_BRANCH}] (or switch to it if it already exists)"
+git checkout -b ${STAGING_BRANCH} ${REMOTE_NAME}/${STAGING_BRANCH} || git checkout ${STAGING_BRANCH}
```

Comment:
Post-merge comment: Can you add a reset if it fallback to the right of the `||`?

We want to be sure that the existent branch is on par with the remote.

(on the following steps where you checkout `CURRENT_BRANCH`, it doesn't matter because you end doing a hard reset to replace everything)

Reply:
good catch - fixed here: https://github.com/kiali/kiali.io/pull/442